### PR TITLE
feat: Badge API

### PR DESCRIPTION
**Pull request by MeguminSama**
- Moved all authors into Devs object
- Badges!

### CODE REVIEW

File: src/api/Badges.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { User } from "discord-types/general";
+import { HTMLProps } from "react";
+
+export enum InsertAt {
```
Comment (Vendicated) [inline src/api/Badges.ts:22]:
```suggestion
export enum BadgePosition {
```

File: src/api/Badges.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    /** The custom image to use */
+    image?: string;
+    /** Action to perform when you click the badge */
+    onClick?(): void;
+    /** Should the user display this badge? */
+    shouldShow?(userInfo: BadgeUserArgs): boolean;
+    /** Optional props (e.g. style) for the badge */
+    props?: HTMLProps<HTMLImageElement>;
+    /** Insert at start or end? */
+    insertAt?: InsertAt;
```
Comment (Vendicated) [inline src/api/Badges.ts:39]:
```suggestion
    position?: BadgePosition;
```

File: src/api/Badges.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+export function addBadge(badge: ProfileBadge) {
+    Badges.add(badge);
+}
+
+/**
+ * Unregister a badge from the Badges API
+ * @param badge The badge to remove
+ */
+export function removeBadge(badge: ProfileBadge) {
+    Badges.delete(badge);
```
Comment (Vendicated) [inline src/api/Badges.ts:60]:
```suggestion
    return Badges.delete(badge);
```

**Review by Vendicated — CHANGES_REQUESTED** [review]

**Comment by Vendicated** [comment]
pls rename file to apiBadges or something like that to have all api plugins show grouped!!!

**Review by Vendicated — APPROVED** [review]

**Comment by ArjixWasTaken** [comment]
> pls rename file to apiBadges or something like that to have all api plugins show grouped!!!

Isn't it time we move all the internal patches to a separate folder?

**Comment by Vendicated** [comment]
that would require a refactor of the commands module builder which I'm too lazy to do rn

**Pull request merged**